### ISSUE TITLE
Max safe integer not compatible with graphql type

### ISSUE THREAD

**Issue by chladog** [vendure-ecommerce/vendure#696]
**Describe the bug**
in getSaleableStockLevel method there is this piece of code:
```
        const inventoryNotTracked =
            variant.trackInventory === GlobalFlag.FALSE ||
            (variant.trackInventory === GlobalFlag.INHERIT && trackInventory === false);
        if (inventoryNotTracked) {
            return Number.MAX_SAFE_INTEGER;
        }
```
Which returns max JS safe integer, however that is not compatible with GraphlQLs "Int" type. _(Int cannot represent non 32-bit signed integer value: 9007199254740991)_

**Expected behavior**
Max number possible for GraphQL Int type.

**Environment (please complete the following information):**
 - @vendure/core version: v0.18.4

**Comment by michaelbromley** [comment]
Is the graphql error occurring in one of the default API operations? Or are you using that method in a custom operation?

**Comment by chladog** [comment]
I used this method for custom "saleable" resolver.

**Comment by michaelbromley** [comment]
I considered changing it to:
```ts
return 2 ** 31 -1; // max 32-bit signed int, so it is safe for the GraphQL Int type
```
but I decided against it because I don't want the GraphQL abstraction leaking into a service method like that. Since you control the resolver, it's trivial to add a `Math.min()` around the result to make it safe for GraphQL.

On a related note, I've just implemented a solution to #442 which will allow you to specify a `StockDisplayStrategy` which you may find preferable to use in future.
